coriolis/tests/osmorphing/netpreserver/test_netplan.py

```diff
@@ -0,0 +1,150 @@
+
+        result = self.netpreserver.check_net_preserver()
+
+        self.assertFalse(result)
+
+    @mock.patch.object(debian.BaseDebianMorphingTools, '_read_file_sudo')
+    @mock.patch.object(debian.BaseDebianMorphingTools, '_list_dir')
+    def test_parse_network(self, mock_list_dir, mock_read_file_sudo):
+        mock_list_dir.return_value = ["01-netplan.yaml"]
+        yaml_content = """
+            network:
+            ethernets:
+                eth0:
+                    match:
+                        macaddress: "00:11:22:33:44:55"
+                    addresses:
+                        - "192.168.1.10/24"
+                eth1:
+                    addresses:
+                        - "192.168.1.20/24"
+                        - "192.168.1.21/24"
+                eth2:
+                    match:
+                        macaddress: "AA:BB:CC:DD:EE:FF"
+                eth4:
+                    addresses:
+                        - "192.168.1.31/24":
+                eth3:
+                    addresses:
+                        - "192.168.1.30/24":
+                            lifetime: 0
+                            label: "eth3"
+        """
+        mock_read_file_sudo.return_value = yaml_content.lstrip().encode()
+
+        self.netpreserver.parse_network()
+
+        expected_info = {
+            "eth0": {
+                "mac_address": "00:11:22:33:44:55",
+                "ip_addresses": ["192.168.1.10"]
+            },
+            "eth1": {
+                "mac_address": "",
+                "ip_addresses": ["192.168.1.20", "192.168.1.21"]
+            },
+            "eth2": {
+                "mac_address": "AA:BB:CC:DD:EE:FF",
+                "ip_addresses": []
+            },
+            "eth3": {
+                "mac_address": "",
+                "ip_addresses": ["192.168.1.30"]
+            },
+            "eth4": {
+                "mac_address": "",
+                "ip_addresses": ["192.168.1.31"]
+            }
+        }
+
+        self.assertEqual(self.netpreserver.interface_info, expected_info)
+
+    @mock.patch.object(debian.BaseDebianMorphingTools, '_read_file_sudo')
+    @mock.patch.object(debian.BaseDebianMorphingTools, '_list_dir')
+    def test_parse_network_invalid_file(self, mock_list_dir,
+                                        mock_read_file_sudo):
+        mock_list_dir.return_value = ["01-netplan.yaml"]
+        yaml_content = """
+            network:
+            ethernets:
+                eth0:
+                        match:
+                    macaddress: "00:11:22:33:44:55"
+                    addresses:
+                        - "192.168.1.10/24"
+        """
+
+        mock_read_file_sudo.return_value = yaml_content.lstrip().encode()
+
+        with self.assertLogs(level='WARNING'):
+            self.netpreserver.parse_network()
+
+        expected_info = {}
+
+        self.assertEqual(self.netpreserver.interface_info, expected_info)
```

Comment:
Please also add the following cases:
Have an empty object as an item for `addresses`;
```yaml
- addresses:
  - {}
  - 192.168.0.24
```
Have an interface with no addresses or empty addresses list.